autogen/main.tf

```diff
@@ -261,6 +261,17 @@ locals {
     regional = element(concat(google_container_cluster.primary.*.vertical_pod_autoscaling.0.enabled, [""]), 0)
     zonal    = element(concat(google_container_cluster.zonal_primary.*.vertical_pod_autoscaling.0.enabled, [""]), 0)
   }
+
+  cluster_type_workload_identity_namespace = {
+    regional = element(concat(google_container_cluster.primary.*.workload_identity_config.0.identity_namespace, [""]), 0)
+    zonal    = element(concat(google_container_cluster.zonal_primary.*.workload_identity_config.0.identity_namespace, [""]), 0)
```

Comment:
On doing some additional testing, this might need to be rethinked a bit more. Going from a WI enabled cluster to empty list (on another note you can't disable WI on a cluster via terraform without forcing a cluster replacement, https://github.com/terraform-providers/terraform-provider-google/issues/4041) causes terraform errors:

```
Error: Invalid index

  on .terraform/modules/gke/modules/beta-private-cluster/main.tf line 249, in locals:
 249:     regional = element(concat(google_container_cluster.primary.*.workload_identity_config.0.identity_namespace, [""]), 0)

The given key does not identify an element in this collection value.
```

Reply:
Yeah, we might consider not having an output for now if you want a quick fix.

Reply:
Removed from output/local vars @morgante